### PR TITLE
Add org.kde.skanpage

### PR DESCRIPTION
**Pull request by JBBgameich**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. *(I am a KDE developer in general, not specifically in this project)*
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned.

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
cc @flathub/kde

File: org.kde.skanpage.json [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=all",
+        "--share=network",
+        "--system-talk-name=org.freedesktop.Avahi"
+    ],
+    "modules": [
+        {
+            "name": "libusb1",
```
Comment (hfiguiere) [inline org.kde.skanpage.json:17]: you should use the one in shared-modules

File: org.kde.skanpage.json [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    "rename-icon": "skanpage",
+    "command": "skanpage",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=all",
+        "--share=network",
+        "--system-talk-name=org.freedesktop.Avahi"
+    ],
```
Comment (bbhtt) [inline org.kde.skanpage.json:15]: It's missing any cleanups. There are lots of stuff to clean up from the final build.

Comment (bbhtt) [reply org.kde.skanpage.json:15]:
```
- "*.a"
- "*.la"
- /include
- /lib/cmake
- /lib/pkgconfig
- /mkspecs
- /share/doc
- /share/man
- /share/qlogging-categories5
```

There's an empty `/app/var/lock` folder being created.

**Comment by Justinzobel** [comment]
Last build is 404, bot, build

**Comment by Justinzobel** [comment]
bot, build org.kde.skanpage

File: org.kde.skanpage.json [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+                    "x-checker-data": {
+                        "type": "anitya",
+                        "project-id": 4954,
+                        "stable-only": true,
+                        "url-template": "https://github.com/tesseract-ocr/tesseract/archive/refs/tags/$version.tar.gz"
+                    }
+                },
+                {
+                    "type": "file",
+                    "url": "https://github.com/tesseract-ocr/tessdata_best/raw/4.1.0/eng.traineddata",
```
Comment (tsdgeos) [inline org.kde.skanpage.json:154]: There's a gazillion more languages supported by tesseract, what was your rationale for including this exact subset?

**Comment by Justinzobel** [comment]
Does flatpak support app add-ons?

**Comment by hfiguiere** [comment]
yes it does.

There is this package doing it https://github.com/flathub/com.github.unrud.djpdf/blob/master/com.github.unrud.djpdf.yaml

**Comment by hfiguiere** [comment]
(for that specific use case)

File: org.kde.skanpage.json (hunk 3 above)
Comment (travier) [reply org.kde.skanpage.json:154]: https://github.com/flathub/org.kde.skanpage/issues/1

**Pull request closed without merging**